internal/validation/validation.go

```diff
@@ -701,22 +702,50 @@ func validateLiteral(c *opContext, l common.Literal) {
 	}
 }
 
+func stringLiteral(value string) common.Literal {
+	return &common.BasicLit{Type: scanner.String, Text: value}
+}
+
 func parseLiteral(value string) common.Literal {
```

Comment:
@thaddock AFAIR this method was called `parseLiteral` in the original code, so i don't feel we should introduce more diff from the master repository by renaming it anyhow

Reply:
@xmirya It's not the original code, I've introduced this function to be able to validate variable values stored in dicts. Perhaps something like `parseWhole` would indeed be a more descriptive name.

Reply:
I have moved panic handling inside `parseLiteral()`, see edit. Currently the only alternative I see is to reimplement graphql's literal parsing, otherwise it can easily throw on simple plain strings that don't necessarily parse as valid graphql code. See the last test case in test_enum.go, without panic recovery in `parseLiteral()` it throws the same panic.

Reply:
Currently, `varBinding` return values of scalar types, or GraphQL literals that are stored in `Variables` dictionary as strings/bytearrays. Slices and dicts should probably be considered as well, but a proper implementation would need `reflect`, wouldn't it?